realm/realm-library/src/main/java/io/realm/ManagedSetManager.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import io.realm.internal.Freezable;
+import io.realm.internal.ManageableObject;
+
+/**
+ * TODO
+ *
+ * @param <E>
+ */
+public class ManagedSetManager<E> implements Set<E>, ManageableObject, Freezable<RealmSet<E>> {
```

Comment:
It is a bit unclean what the purpose of this is. Seems like the strategy is already used as indirection. Do we really also want the `ManagedStrategy` to have another layer inside or?

Reply:
True, I was on autopilot when I added this. I was basing the architecture on dictionaries, and there it is needed as we now have `RealmDictionary` and in the future could have other types, but here this abstraction is not necessary. But don't worry, I realised it was superfluous later on and removed it.